### DISCUSSION TITLE
Beginner question

### DISCUSSION THREAD

**Discussion by da4089** [da4089/simplefix#60] [Q&A]
(from @simobuckz, moved from a commit comment)

Heyy im new to Github, ive always had it. I bought this book on algo trading. is there anyway that you can guide me. like where can I start on chapter one. which folder do I use. If you like to help me(PLEASE!) You can email me at <redacted> I will greatly appreciate your help sir. Thank you so much. God bless

**Comment by da4089** [comment]
@simobuckz, I don't understand what you're asking, sorry.

Your mention of "which folder do I use" suggests that perhaps you're not very familiar with Python?  In which case, you should find many, many online tutorials for it.  I don't know if this one is any better than any other, but I've heard of it before ... https://diveintopython3.net/

For FIX protocol, there aren't as many introductory resources, but I found this which looks like a good starting point ... https://www.fixsim.com/fix-protocol-tutorial

For basic use of _simplefix_, see the [README](https://github.com/da4089/simplefix/blob/master/README.rst) for a quick introduction to the module, and then the full documentation ... https://simplefix.readthedocs.io/en/latest/

I cannot offer advice on algorithmic trading, but again, a search engine will offer numerous resources.

Best of luck!

**Comment by simobuckz** [comment]
Thank you for timely response. Im somewhat familiar with python but not so much. Just the basics. This will be my first project I’m engaging in. It was the first language I learned. I just wanted to know how to start using the codes that are given for the project. I already downloaded Git and Homebrew. I already have Vs code on my external hard drive. I searched up some videos on how I can start using Git.  I can either use VSC to code, I just have to create a repository, and then I can start creating right? Or I can use my terminal. Thats my understanding so far.

Thank you in advance

On Jun 23, 2024, at 5:57 PM, David Arnold ***@***.***> wrote:

You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

[WARNING] This message comes from an external organization. Be careful of embedded links and attachments.



@simobuckz<https://github.com/simobuckz>, I don't understand what you're asking, sorry.

Your mention of "which folder do I use" suggests that perhaps you're not very familiar with Python? In which case, you should find many, many online tutorials for it. I don't know if this one is any better than any other, but I've heard of it before ... https://diveintopython3.net/

For FIX protocol, there aren't as many introductory resources, but I found this which looks like a good starting point ... https://www.fixsim.com/fix-protocol-tutorial

For basic use of simplefix, see the README<https://github.com/da4089/simplefix/blob/master/README.rst> for a quick introduction to the module, and then the full documentation ... https://simplefix.readthedocs.io/en/latest/

I cannot offer advice on algorithmic trading, but again, a search engine will offer numerous resources.

Best of luck!

**Comment by da4089** [comment]
Homebrew, Git, and VSCode are all reasonable choices.

You will find an enormous number of Git tutorials with a search engine, most of which will cover the mechanics of creating a new project.  While using Git is definitely a good idea, note that it's also not strictly necessary.

I almost always recommend using a Python virtual environment (aka venv) for development.  Again, there are countless Python tutorials that explain how to accomplish this, and make recommendations about directory structures, etc.

Then, as explained in the README, you should install _simplefix_ from PyPI.  The default tool for doing this is _pip_, although other options exist if you want to use them for some reason.

Having done that, you can begin the development of your application.

**BUT** ...

To be honest, based on this discussion, I would suggest that you consider a series of simpler projects first.  An algorithmic trading system is not a trivial project.  Most projects, if they have bugs, won't lose you large amounts of money, but a trading algorithm might.  Please consider developing your skills further before you embark on this task.
